.github/workflows/main.yaml

```diff
@@ -8,37 +9,52 @@ on:
 jobs:
   gomod:
     runs-on: ubuntu-20.04
+    outputs:
+      gomod: ${{ steps.gomod.outputs.gomod }}
+      gosum: ${{ steps.gosum.outputs.gosum }}
     steps:
     - uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4.1.1
     - run: make go-mod-tidy
-    - run: make go-mod-download
-    - run: tar -cvf ./src.tar.gz ./ # preserve file permissions
-    - uses: actions/upload-artifact@c7d193f32edcb7bfad88892161225aeda64e9392 # v4.0.0
-      with:
-        name: src
-        path: ./src.tar.gz
+    - id: gomod
+      run: |
+        {
+          echo 'gomod<<FILE'
+          cat go.mod
+          echo
+          echo FILE
+        } >> "$GITHUB_OUTPUT"
+    - id: gosum
+      run: |
+        {
+          echo 'gosum<<FILE'
+          cat go.sum
+          echo
+          echo FILE
+        } >> "$GITHUB_OUTPUT"
```

Comment:
earlier we used to upload the code from one step and download it on other step? How does that work now? I see we are just outputting the go.* files and getting them from other step. How is code available in other steps?

Also, it looks like we are not running `go mod download` anymore, do we not need that?

Last question :smile: , why do we have to just copy the go.* between steps and not other code.

Reply:
We do have the `checkout` step, which updates the code. `go.` files were synced before, but I'm not sure that's strictly required TBH. We could make it a check (that all deps are up to date) and don't move the `go.` files between jobs, which would make sense, but I just didn't want to introduce new behaviours in this PR. 
Doing `go mod download` and then packing and unpacking the files is way longer than to let each step (lint, build) to download deps themselves. Also if you run `go mod tidy` it does download all dependencies and there is no need to run `go mod download` again.

Reply:
Let me join the questions left by @viveksinghggits . I'd also put some textual notes to the workflow to explain why we are doing that / what we are trying to achieve.